src/overlays/actors/ovl_En_Pst/z_en_pst.c

```diff
@@ -31,51 +108,311 @@ const ActorInit En_Pst_InitVars = {
+            return gSaveContext.save.weekEventReg[27] & 0x20;
+        default:
+            return false;
+    }
+}
+
+s32 func_80B2B8F4(EnPst* this) {
+    switch (this->actor.params) {
+        case 0:
+            return gSaveContext.save.weekEventReg[27] & 0x40;
+        case 1:
+            return gSaveContext.save.weekEventReg[27] & 0x80;
+        case 2:
+            return gSaveContext.save.weekEventReg[28] & 0x1;
+        case 3:
+            return gSaveContext.save.weekEventReg[28] & 0x2;
+        case 4:
+            return gSaveContext.save.weekEventReg[28] & 0x4;
+        default:
+            return false;
+    }
+}
+
+s32 EnPst_ChooseBehaviour(Actor* thisx, PlayState* play) {
+    s32 itemActionParam = 0;
+    s32 scriptBranch = 0;
+    EnPst* this = THIS;
+
+    switch (this->behaviour) {
+        case POSTBOX_BEHAVIOUR_WAIT_FOR_ITEM:
+            switch (Message_GetState(&play->msgCtx)) {
+                case TEXT_STATE_CHOICE:
+                case TEXT_STATE_5:
+                    if (Message_ShouldAdvance(play)) {
+                        case TEXT_STATE_16:
+                            itemActionParam = func_80123810(play);
+                            scriptBranch = 0;
+                            if ((itemActionParam == PLAYER_AP_LETTER_TO_KAFEI) ||
+                                (itemActionParam == PLAYER_AP_LETTER_MAMA)) {
+                                this->exchangeItemId = itemActionParam;
+                                this->behaviour++;
+                                scriptBranch = 1;
+                            } else if (itemActionParam < PLAYER_AP_NONE) {
+                                this->behaviour++;
+                                scriptBranch = 3;
+                            } else if (itemActionParam != PLAYER_AP_NONE) {
+                                this->behaviour++;
+                                scriptBranch = 2;
+                            }
+                    }
+                    break;
+            }
+            break;
+        case POSTBOX_BEHAVIOUR_TAKE_ITEM:
+            if (this->exchangeItemId == EXCH_ITEM_LETTER_TO_KAFEI) {
+                scriptBranch = 1;
+            }
+            break;
+    }
+    return scriptBranch;
+}
+
+s32* EnPst_GetMsgEventScript(EnPst* this, PlayState* play) {
+    if (Player_GetMask(play) == PLAYER_MASK_POSTMAN) {
+        return D_80B2C3B8;
+    }
+
+    if (this->stateFlags & 0x10) {
+        switch (this->actor.params) {
+            case 0:
+                return D_80B2C3E8;
+            case 1:
+                return D_80B2C408;
+            case 2:
+                return D_80B2C428;
+            case 3:
+                return D_80B2C448;
+            case 4:
+                return D_80B2C468;
+            default:
+                return NULL;
+        }
+
+    } else if (this->stateFlags & 0x20) {
+        if (this->exchangeItemId == 0x33) {
+            return D_80B2C488;
+        }
+        return D_80B2C490;
```

Comment:
```suggestion
            return D_80B2C488;
        } else
            return D_80B2C490;
        }
```
Just looks a bit neater.